### ISSUE TITLE
bug: `AllowedHeaders` in S3 CORS configurations should not be case-sensitive

### ISSUE THREAD

**Issue by bigen1925** [localstack/localstack#9957]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

`AllowedHeaders` in S3 CORS configurations seems to be case-sensitive.

For example, 
```
{
  "CORSRules": [
    {
      "AllowedHeaders": ["Content-Type"],
      ...
    }
  ]
}
```
this settings reject pre-flight requests with
```
Access-Control-Request-Headers: content-type
```

### Expected Behavior

Access-Control-Request-Headers of preflight requests is accepted in case-insensitive as real S3 does so.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker compose up

```yaml
  localstack:
    image: localstack/localstack:3.0.2
    volumes:
      - ./localstack/ready.d:/etc/localstack/init/ready.d
      - /var/run/docker.sock:/var/run/docker.sock
    ports:
      - '4566:4566'
    environment:
      AWS_ACCESS_KEY_ID: test
      AWS_SECRET_ACCESS_KEY: test
      REGION: ap-northeast-1
      S3_BUCKET: bucket
      S3_SKIP_SIGNATURE_VALIDATION: 0
      S3_CORS_ALLOWED_ORIGIN: http://localhost:3000 // frontend application port
```

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

```
# make bucket
awslocal s3 mb s3://${S3_BUCKET}

# cors settings
awslocal s3api put-bucket-cors --bucket ${S3_BUCKET} --cors-configuration '
{
  "CORSRules": [
    {
      "AllowedHeaders": ["Content-Type"],
      "AllowedMethods": ["GET", "PUT"],
      "AllowedOrigins": ["'http://localhost:3000'"]
    }
  ]
}
'
```

#### And then

1. generate signed url of putObject
2. putObject from browser application
```
fetch(signedUrl, { method: 'PUT', body: file })
```

fetch API sends pre-flight requests with 
```
Access-Control-Request-Headers: content-type
```
in default (Note that `content-type` is with all lower cases thought CORS settings is with Upper cases), and it is rejected.

### Environment

```markdown
- OS: Mac OS 14.2.1
- LocalStack: 3.0.2
```

### Anything else?

I tried to put CORS configurations with 
```
"AllowedHeaders": ["content-type"],  // (not "Content-Type)
```
, then pre-flight requests are accepted.

-----

This is a screen shot of a rejected pre-flight request

<img width="400" alt="スクリーンショット 2023-12-29 10 47 04" src="https://github.com/localstack/localstack/assets/25700049/7b83f671-67e5-404f-927f-0c2a194e6de7">
<img width="400" alt="スクリーンショット 2023-12-29 10 47 13" src="https://github.com/localstack/localstack/assets/25700049/03864112-848c-405b-ad79-6ce61489b754">

**Comment by bentsku** [comment]
Thanks a lot for reporting! This should be fixed with #9973 which has just been merged. The fix will be contained in the `latest` tag of the Docker image `localstack/localstack`, which should be available in the next hour. Thanks again for the report!

**Comment by bigen1925** [comment]
@bentsku 
Thank you so much for your quick fix!

I confirmed that the issue was fixed on `latest` image! 😃 

I love and appreciate for all developers of `localstack`.

Have a nice holiday!
